pkg/webhook/ingress/validating.go

```diff
@@ -120,9 +119,30 @@ func (r *handler) validateIngress(ctx context.Context, c client.Client, object I
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 
-	if !tl.Items[0].Spec.IngressClasses.IsStringInList(*ingressClass) {
+	if len(tl.Items[0].Spec.IngressClasses.AllowedRegex) > 0 || len(tl.Items[0].Spec.IngressClasses.Allowed) > 0 {
+		ingressClassEnforced = true
+	}
+
+	if !ingressClassEnforced {
```

Comment:
I guess we have to change the behavior, here: we have to cover multiple cases, like denying any kind of Ingress or PVC creation with the following YAML definition:

```
# deny any Ingress
ingressClasses:
  allowed: []
  allowedRegex: ""
```

```
# allow any Ingress
ingressClasses:
  allowed: []
  allowedRegex: .*
```

So all the business logic here should be changed according to this, otherwise would be impossible to achieve also this kind of expected behavior.

Reply:
done